osa/provenance/utils.py

```diff
@@ -62,69 +70,92 @@ def parse_variables(class_instance):
             calibration_path / timecalibration_filename
         )
         class_instance.PointingFile = class_instance.args[3]
-        class_instance.ObservationRun = class_instance.args[5].split(".")[0]
-        class_instance.ObservationSubRun = class_instance.args[5].split(".")[0]
+        class_instance.RunSummaryFile = os.path.basename(class_instance.args[4])
+        class_instance.DL1SubrunDataset = (
+            f"{outdir_dl1}/dl1_LST-1.Run{class_instance.args[5]}.h5"
+        )
+
+    if class_instance.__name__ == "dl1ab":
+        # run_str       [0] 02006.0000
+
+        class_instance.Analysisconfigfile_dl1 = configfile_dl1
+        class_instance.ObservationRun = class_instance.args[0].split(".")[0]
         class_instance.ObservationDate = nightdir
         class_instance.SoftwareVersion = options.lstchain_version
-        class_instance.ProdID = options.prod_id
-        class_instance.CalibrationRun = re.findall(
-            r"Run(\d{5}).", calibration_filename
-        )[0]
-        class_instance.PedestalRun = re.findall(
-            r"Run(\d{5}).", pedestal_filename
-        )[0]
-        outdir_dl1 = Path(dl1_dir) / nightdir / options.prod_id
+        class_instance.session_name = class_instance.ObservationRun
+        class_instance.ProcessingConfigFile = options.configfile
+
+        class_instance.PedestalCleaning = "True"
+        class_instance.StoreImage = cfg.getboolean("lstchain", "store_image_dl1ab")
+
         class_instance.DL1SubrunDataset = (
-            f"{outdir_dl1}{dl1_prefix}.Run{class_instance.args[5]}.h5"
+            f"{outdir_dl1}/dl1_LST-1.Run{class_instance.args[0]}.h5"
         )
-        # /fefs/aswg/data/real/R0/20200218/LST1.1.Run02006.0001.fits.fz
-        class_instance.R0SubrunDataset = f"{rawdir}/" \
-                                         f"{class_instance.ObservationDate}/" \
-                                         f"{r0_prefix}." \
-                                         f"Run{class_instance.args[5]}.fits.fz"
+
+    if class_instance.__name__ == "dl1_datacheck":
+        # run_str       [0] 02006.0000
+
+        class_instance.ObservationRun = class_instance.args[0].split(".")[0]
+        class_instance.ObservationDate = nightdir
+        class_instance.SoftwareVersion = options.lstchain_version
         class_instance.session_name = class_instance.ObservationRun
         class_instance.ProcessingConfigFile = options.configfile
 
+        class_instance.DL1SubrunDataset = (
+            f"{outdir_dl1}/dl1_LST-1.Run{class_instance.args[0]}.h5"
+        )
+        class_instance.MuonsSubrunDataset = (
+            f"{analysis_dir}/muons_LST-1.Run{class_instance.args[0]}.fits"
```

Comment:
{outdir_dir}

Reply:
Actually, these muons files are produced in the previous step `r0_to_dl1` so I guess `MuonsSubrunDataset` should be moved to the previous class instance.

Reply:
Ok.
Since these muons files are produced by `r0_to_dl1` I will add a `wasGeneratedBy` connection, keeping also the already existing `used` connection in `dl1_datacheck`.